core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -88,8 +88,8 @@ public RecordState validateTransition(RecordState newState) {
                 throw new IllegalStateException("The state transition is invalid from the current state: " + this);
             }
 
-            if (this == AVAILABLE && newState != ACQUIRED) {
-                throw new IllegalStateException("The state can only be transitioned to ACQUIRED from AVAILABLE");
+            if (this == AVAILABLE && newState == ACKNOWLEDGED) {
```

Comment:
Can we have a transition from AVAILABLE to ARCHIVED? I don't think so.

Reply:
Thanks for the PR review. My understanding of the existing code was a bit off. You are right here, I have updated the code in the latest commit.